packages/concerto-core/lib/decoratorutil.js

```diff
@@ -0,0 +1,301 @@
+                extractionDictionary = DecoratorUtil.constructDCSDictionary(extractionDictionary, model.namespace, model.decorators, {});
+                if (removeDecoratorsFromModel){
+                    model.decorators = undefined;
+                }
+            }
+            const processedDecl = model.declarations.map(decl => {
+                if (decl.decorators) {
+                    const constructOptions = {
+                        declaration: decl.name,
+                    };
+                    extractionDictionary = DecoratorUtil.constructDCSDictionary(extractionDictionary, model.namespace, decl.decorators, constructOptions);
+                }
+                if (removeDecoratorsFromModel){
+                    decl.decorators = undefined;
+                }
+                if (decl.$class.endsWith('.MapDeclaration')) {
+                    if (decl.key){
+                        if (decl.key.decorators){
+                            const constructOptions = {
+                                declaration: decl.name,
+                                mapElement: 'KEY'
+                            };
+                            extractionDictionary = DecoratorUtil.constructDCSDictionary(extractionDictionary, model.namespace, decl.key.decorators, constructOptions);
+                            decl.key.decorators = undefined;
+                        }
+                    }
+                    if (decl.value){
+                        if (decl.value.decorators){
+                            const constructOptions = {
+                                declaration: decl.name,
+                                mapElement: 'VALUE'
+                            };
+                            extractionDictionary = DecoratorUtil.constructDCSDictionary(extractionDictionary, model.namespace, decl.value.decorators, constructOptions);
+                            decl.value.decorators = undefined;
+                        }
+                    }
+                }
+                if (decl.properties) {
+                    const processedProperties = decl.properties.map(property => {
+                        if (property.decorators){
+                            const constructOptions = {
+                                declaration: decl.name,
+                                property: property.name
+                            };
+                            extractionDictionary = DecoratorUtil.constructDCSDictionary(extractionDictionary, model.namespace, property.decorators, constructOptions );
+                        }
+                        if (removeDecoratorsFromModel){
+                            property.decorators = undefined;
+                        }
+                        return property;
+                    });
+                    decl.properties = processedProperties;
+                }
+                return decl;
+            });
+            model.declarations = processedDecl;
+            return model;
```

Comment:
What I meant before, was that this huge chunk is collection and can be separated out to a different function. Having everything in this function, simplified `extractDecorators` but now this huge chunk is under this function.